tests/unit/test_user.py

```diff
@@ -26,6 +26,37 @@ def test_full_name(empty_test_db, create_admin_user):
     assert admin.full_name() == "first last"
 
 
+@pytest.mark.usefixtures("empty_test_db")
+class TestFindUsersWithoutAssignedRole:
+    def test_find_users_authorized_active(self, create_admin_user):
+        _ = create_admin_user()
+
+        found_user = UserModel.find_users_without_assigned_role()
+        assert found_user.first() is None
+
+    def test_find_users_authorized_inactive(self, create_admin_user):
+        user = create_admin_user()
+        user.archived = True
+        user.save_to_db()
+
+        found_user = UserModel.find_users_without_assigned_role()
+        assert found_user.first() is None
+
+    def test_find_users_unauthorized_active(self, create_unauthorized_user):
+        user = create_unauthorized_user()
+
+        found_user = UserModel.find_users_without_assigned_role()
+        assert found_user.first() == user
```

Comment:
I didn't know about this `first()` method. That is pretty cool. And there shouldn't be a second result, because we only created one user. Hmmm. It looks like there is a `.all()` method - https://flask-sqlalchemy.palletsprojects.com/en/2.x/queries/?highlight=first - I would prefer to use the .all() method in all of the tests.
`assert result.all() == [user]`
and instead of testing for `None` we test the query is empty.
`assert result.all() == []`
Because the query doesn't ever return `None` so I don't think we should be testing for it.